implement-directus-auth-in-nextjs-14/index.md

```diff
@@ -0,0 +1,1105 @@
+
+![Directus public role field validation settings](public-role-field-validation-settings.png)
+
+Click on `Field Permission` and check the fields required to create a user.
+
+![Directus public role field permission settings](public-role-field-permission-settings.png)
+
+The required fields are:
+
+- first_name
+- last_name
+- email
+- password
+- role
+
+**Creating posts collection**
+Navigate to `Settings/Data Model` and create a new collection named `posts`. This is where all the posts created by users with an Author role will be stored.
+
+![Directus posts collection setup](Directus-posts-collection-setup.png)
+
+Check the fields that you want returned on the `posts` data
+
+![Directus posts optional fields settings](Directus-posts-optional-fields-settings.png)
+
+Create additional fields for the post:
+
+- title => Input field
+- content => WYSIWYG
+- image => Image
+
+The `title` and `content` fields should be required.
+
+![Directus post additional fields settings](![alt text](Directus-post-additional-fields-settings-1.png)
+
+## **Initialize Next.js Project**
+
+Open your terminal and run the following command to create a new Next.js project:
+
+```bash
+npx create-next-app@latest
+```
+
+On installation, choose the following configurations when prompted:
+
+```text
+What is your project named? next-blog
+Would you like to use TypeScript? No
+Would you like to use ESLint? Yes
+Would you like to use Tailwind CSS? No
+Would you like to use `src/` directory? Yes
+Would you like to use App Router? Yes
+Would you like to customize the default import alias (@/*)? Yes
+```
+
+After the prompts, `create-next-app` will create a folder with your project name and install the required dependencies.
+
+Install the required dependencies:
+
+```bash
+    npm i next-auth @directus/sdk
+```
+
+## Set Up the ENV Variables
+
+Create a `.env` file in your Next.js project root directory and add the following:
+
+```text
+NEXTAUTH_URL=http://localhost:3000
+NEXTAUTH_SECRET=YOUR_NEXT_RANDOM_SECRET
+USER_ROLE=USER_ROLE_ID_FROM_YOUR_DIRECTUS_PROJECT
+BACKEND_URL=YOUR_DIRECTUS_PROJECT_URL
+NEXT_PUBLIC_URL=http://localhost:3000
+```
+
+You can access your Directus project from the Next.js application via `REST`, `GraphQL` and `SDK`. 
+
+## Set Up the Directus SDK
+
+Create an instance of the Directus SDK that will be used across the next.js application.
+In the `next-blog` project `src` directory, create a new directory called `lib` and a new file called `directus.js` inside `lib`, then add the following code:
+
+```javascript
+import { createDirectus, rest, authentication } from '@directus/sdk';
+const directus = createDirectus(process.env.BACKEND_URL)
+  .with(authentication("cookie", {credentials: "include", autoRefresh: true}))
+export default directus;
+```
+
+This code snippet sets up a client for interacting with the Directus backend with cookie-based authentication, ensuring that credentials are included in requests and enabling automatic token refresh.
```

Comment:
Remove all of this in favor of linking to this document: https://docs.directus.io/guides/headless-cms/build-static-website/next

This post is meant to just be about auth. This has the added benefit of having some collections already set up. Make this a requirement in the **Before You Start**.

Obviously, you won't be able to add the `USER_ROLE` env var yet, but that's ok. Make that a step at the end of the "Creating a New Role" section.